test/pylib/util.py

```diff
@@ -131,6 +130,13 @@ async def find_host():
     return await wait_for(find_host, deadline)
 
 
+async def safe_gather(*awaitables: Awaitable, return_exceptions: bool = True):
```

Comment:
Please add a comment explaining what is "safe_gather", and what makes the usual asyncio.gather unsafe, and what the return_exceptions parameter means and when it should be given the non-default value false (below @temichus suggested it shouldn't even be allowed to be false?)

Reply:
Given the PR description I don't fully grasp the problem. Is it because when first exception is raised and propagated the other tasks from gather are still running?

perhaps it'd be enough to wait in the tear-down until they complete using asyncio.all_tasks ? Assuming that we don't create those tasks endlessly  somewhere in the background.

Reply:
@nyh returntException allows us to treat exceptions the same way as a success result so that it will be returned in the final list. Without it exceptions will be propagated up in the task will be cancelled.  We are using the nested gather() method so we can have issues with race conditions. There is an alternative asyncio.TaskGroup but it's appeared only in 3.11.

Reply:
@nuivall In both cases the rest of the tasks are not canceled, but without returnExceptions it will be propagated immediately to the caller, with returnExceptions will be added as a result. Potentially it will lead to increasing time of execution but only when we have fails.

Reply:
> Potentially it will lead to increasing time of execution but only when we have fails.

If in both cases all tasks need to run to completion how one case can be faster than another?

A localized solution (i.e. change in one place) if possible would be preferred than adding a wrapper, then you also need to teach people to always (or sometimes?) use it.